File: sample/src/main/java/com/qingmei2/sample/zhihu/ZhihuImagePicker.java
package com.qingmei2.sample.zhihu;

import com.qingmei2.rximagepicker.entity.Result;
import com.qingmei2.rximagepicker.entity.sources.Camera;
import com.qingmei2.rximagepicker.entity.sources.Gallery;

import io.reactivex.Observable;

public interface ZhihuImagePicker {

    String KEY_ZHIHU_PICKER_NORMAL = "key_zhihu_picker_theme_normal";
    String KEY_ZHIHU_PICKER_DRACULA = "key_zhihu_picker_theme_dracula";

    @Gallery(viewKey = KEY_ZHIHU_PICKER_NORMAL)
    Observable<Result> openGalleryAsNormal();

    @Gallery(viewKey = KEY_ZHIHU_PICKER_DRACULA)
    Observable<Result> openGalleryAsDracula();

    @Camera
    Observable<Result> openCamera();
}

File: sample/src/main/java/com/qingmei2/sample/zhihu/ZhihuImagePicker.kt
package com.qingmei2.sample.zhihu

import android.content.Context
import com.qingmei2.rximagepicker.entity.Result
import com.qingmei2.rximagepicker.entity.sources.Camera
import com.qingmei2.rximagepicker.entity.sources.Gallery
import com.qingmei2.rximagepicker.ui.ICustomPickerConfiguration
import com.qingmei2.rximagepicker_extension_zhihu.ui.ZhihuImagePickerActivity

import io.reactivex.Observable

interface ZhihuImagePicker {

    @Gallery(componentClazz = ZhihuImagePickerActivity::class)
    fun openGalleryAsNormal(context: Context,
                            config: ICustomPickerConfiguration): Observable<Result>

    @Gallery(componentClazz = ZhihuImagePickerActivity::class)
    fun openGalleryAsDracula(context: Context,
                             config: ICustomPickerConfiguration): Observable<Result>

    @Camera
    fun openCamera(context: Context): Observable<Result>
}
